pybamm/discretisations/discretisation.py

```diff
@@ -351,16 +351,30 @@ def process_binary_operators(self, bin_op):
             left.has_gradient_and_not_divergence()
             and not right.has_gradient_and_not_divergence()
         ):
+            # Extrapolate at either end depending on the ghost cells (from gradient)
+            extrapolate_left = any(
+                [x.has_left_ghost_cell for x in new_left.pre_order()]
+            )
+            extrapolate_right = any(
+                [x.has_right_ghost_cell for x in new_left.pre_order()]
+            )
             new_right = self._spatial_methods[bin_op.domain[0]].compute_diffusivity(
-                new_right
+                new_right, extrapolate_left, extrapolate_right
```

Comment:
Does this just save us from having to add ghost nodes to new_right?

Reply:
this is for the `NodeToEdge` function to extrapolate appropriately when there are ghost cells